presto-common/src/main/java/com/facebook/presto/common/block/AbstractMapBlock.java

```diff
@@ -190,8 +191,8 @@ public long getRegionLogicalSizeInBytes(int position, int length)
 
         return getRawKeyBlock().getRegionLogicalSizeInBytes(entriesStart, entryCount) +
                 getRawValueBlock().getRegionLogicalSizeInBytes(entriesStart, entryCount) +
-                (Integer.BYTES + Byte.BYTES) * length +
-                Integer.BYTES * HASH_MULTIPLIER * entryCount;
+                (Integer.BYTES + Byte.BYTES) * (long) length +
```

Comment:
do we need the cast to (long) here

Reply:
IDEA was warning about, upcast after the int multiplication has happened. There is possibility that, length * 5 may exceed int32 max and this change will fix that.